src/ekka_rlog_server.erl

```diff
@@ -65,6 +66,15 @@ start_link(Shard) ->
     Config = #{}, % TODO
     gen_server:start_link({local, Shard}, ?MODULE, {Shard, Config}, []).
 
+%% @doc Make a call to the server that does nothing.
+%%
+%% This API function is called by the replicant before `subscribe/3'
+%% to reduce the risk of double subscription when the reply from the
+%% server is lost or delayed due to network congestion.
+-spec probe(node(), ekka_rlog:shard()) -> boolean().
+probe(Node, Shard) ->
```

Comment:
Not a foolproof solution against the double-subscription, but it doesn't make the protocol more complex.